MovieTime/MovieTime/Core/MTCellProtocol.swift

```diff
@@ -0,0 +1,43 @@
+//
+//  MTCellProtocol.swift
+//  MovieTime
+//
+//  Created by Enrique Melgarejo on 09/01/18.
+//  Copyright © 2018 Choynowski. All rights reserved.
+//
+
+import UIKit
+
+protocol MTCellProtocol: class {
+    static var identifier: String { get }
+}
+
+extension MTCellProtocol where Self: UITableViewCell {
```

Comment:
do you have a UITableView in this app?

Reply:
Not yet, but as one of the requirements was to support project growth, I implemented the protocol for 3 types of reusable views:
- UITableViewCell
- UITableViewHeaderFooterView
- UICollectionViewCell

Reply:
honestly sometimes I wonder how code challenges would be if we removed that requirement, because everyone tends to add unnecessary code thinking about growth. What we really want to say with that requirement is that the code needs to be flexible and maintainable, and I think that the fact you have a protocol with an `identifier` already accomplishes that pretty well. You don't need to have unused classes extending that protocol, that doesn't make your project more prepared to the future.